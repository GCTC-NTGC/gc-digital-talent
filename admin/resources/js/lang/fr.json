{
  "+6BCqp": {
    "defaultMessage": "View our terms and conditions.",
    "description": "Title for the terms and conditions link in the Footer."
  },
  "+e8VZ8": {
    "defaultMessage": "Error: updating cmo asset failed",
    "description": "Message displayed to user after cmo asset fails to get updated."
  },
  "+mhbIa": {
    "defaultMessage": "Description EN:",
    "description": "Label displayed on the create a cmo asset form description (English) field."
  },
  "/+IJwt": {
    "defaultMessage": "Create User",
    "description": "Heading displayed above the Create User form."
  },
  "/3W/ui": {
    "defaultMessage": "Operational Requirements:",
    "description": "Label displayed on the pool form operational requirements field."
  },
  "/8nVpX": {
    "defaultMessage": "Name (English):",
    "description": "Label displayed on the create a department form name (English) field."
  },
  "/Y7x+s": {
    "defaultMessage": "Create Pool",
    "description": "Heading displayed above the Create Pool form."
  },
  "/uqLeF": {
    "defaultMessage": "Create User",
    "description": "Title displayed on the create a user form."
  },
  "0+g2jN": {
    "defaultMessage": "Email",
    "description": "Title displayed for the User table Email column."
  },
  "081PsW": {
    "defaultMessage": "Key:",
    "description": "Label displayed on the operational requirement form key field."
  },
  "0SoKjt": {
    "defaultMessage": "User {userId} not found.",
    "description": "Message displayed for user not found."
  },
  "0tYHRO": {
    "defaultMessage": "Select one or more classifications...",
    "description": "Placeholder displayed on the pool form classifications field."
  },
  "154pGu": {
    "defaultMessage": "Users",
    "description": "Label displayed on the Users menu item."
  },
  "1J4e+j": {
    "defaultMessage": "Feedback",
    "description": "Label for the feedback link in the Footer."
  },
  "29OmCu": {
    "defaultMessage": "Expiry",
    "description": "Title displayed for the Pool Candidates table Expiry column."
  },
  "2uslSg": {
    "defaultMessage": "Description FR:",
    "description": "Label displayed on the create a cmo asset form description (French) field."
  },
  "2wmzS1": {
    "defaultMessage": "Name",
    "description": "Title displayed for the Department table Name column."
  },
  "2x34X6": {
    "defaultMessage": "Name (English):",
    "description": "Label displayed on the operational requirement form name (English) field."
  },
  "33VQf6": {
    "defaultMessage": "Select a status...",
    "description": "Placeholder displayed on the pool candidate form status field."
  },
  "3CIhwK": {
    "defaultMessage": "Departments",
    "description": "Label displayed on the Departments menu item."
  },
  "3D/wxC": {
    "defaultMessage": "Classification created successfully!",
    "description": "Message displayed to user after classification is created successfully."
  },
  "3F6QqF": {
    "defaultMessage": "Start writing here...",
    "description": "Placeholder displayed on the Global Filter form Search field."
  },
  "3qnyLY": {
    "defaultMessage": "ID",
    "description": "Title displayed on the Operational Requirement table ID column."
  },
  "43G2B6": {
    "defaultMessage": "Name (French):",
    "description": "Label displayed on the create a department form name (French) field."
  },
  "457hEW": {
    "defaultMessage": "Departments",
    "description": "Heading displayed above the Department Table component."
  },
  "4BAjoi": {
    "defaultMessage": "Description (French):",
    "description": "Label displayed on the pool form description (French) field."
  },
  "4Gf0FC": {
    "defaultMessage": "Select a pool...",
    "description": "Placeholder displayed on the pool candidate form Pool field."
  },
  "4XF9kF": {
    "defaultMessage": "Name",
    "description": "Title displayed for the CMO Asset table Name column."
  },
  "5FFRV2": {
    "defaultMessage": "Error: updating user failed",
    "description": "Message displayed to user after user fails to get updated."
  },
  "5JRGJY": {
    "defaultMessage": "Update CMO Asset",
    "description": "Title displayed on the update a cmo asset form."
  },
  "5jFkhf": {
    "defaultMessage": "Error: creating classification failed",
    "description": "Message displayed to user after classification fails to get created."
  },
  "5lTgcK": {
    "defaultMessage": "Key:",
    "description": "Label displayed on the create a cmo asset form key field."
  },
  "630Eyj": {
    "defaultMessage": "Indigenous:",
    "description": "Placeholder displayed on the pool candidate form is indigenous field."
  },
  "6Lo9lm": {
    "defaultMessage": "Pool Candidate {poolCandidateId} not found.",
    "description": "Message displayed for pool candidate not found."
  },
  "6hONKP": {
    "defaultMessage": "Level:",
    "description": "Label displayed on the classification form level field."
  },
  "7SWIeC": {
    "defaultMessage": "Woman",
    "description": "Title displayed for the Pool Candidates table Woman column."
  },
  "7d/ot8": {
    "defaultMessage": "Toggle All",
    "description": "Label displayed on the Table Columns toggle fieldset."
  },
  "7iglPJ": {
    "defaultMessage": "Create CMO Asset",
    "description": "Heading displayed above the Create CMO Asset form."
  },
  "8AbNBq": {
    "defaultMessage": "Pool Candidates",
    "description": "Heading displayed above the Pool Candidate Table component."
  },
  "8Hw3Zf": {
    "defaultMessage": "User",
    "description": "Title displayed for the Pool Candidates table User column."
  },
  "8Otaw9": {
    "defaultMessage": "Department {departmentId} not found.",
    "description": "Message displayed for department not found."
  },
  "9KSiCg": {
    "defaultMessage": "Visible Minority:",
    "description": "Label displayed on the pool candidate form is visible minority field."
  },
  "9UNZD+": {
    "defaultMessage": "Name (English):",
    "description": "Label displayed on the pool form name (English) field."
  },
  "9c/MAZ": {
    "defaultMessage": "Minimum Salary",
    "description": "Title displayed for the Classification table Minimum Salary column."
  },
  "9nLx6d": {
    "defaultMessage": "Visit Canada.ca.",
    "description": "Title for the Canada link in the Footer."
  },
  "9noaD1": {
    "defaultMessage": "Description (French):",
    "description": "Label displayed on the operational requirement form description (French) field."
  },
  "9zLCSv": {
    "defaultMessage": "Email",
    "description": "Title displayed on the Pool Candidates table email column."
  },
  "ANLJ2T": {
    "defaultMessage": "Pool updated successfully!",
    "description": "Message displayed to user after pool is updated successfully."
  },
  "Ari4Qz": {
    "defaultMessage": "User:",
    "description": "Label displayed on the pool candidate form user field."
  },
  "B0lDYB": {
    "defaultMessage": "Select one or more essential...",
    "description": "Placeholder displayed on the pool form essential criteria field."
  },
  "BEgoJ6": {
    "defaultMessage": "Telephone:",
    "description": "Label displayed on the user form telephone field."
  },
  "BQXfu7": {
    "defaultMessage": "Status:",
    "description": "Label displayed on the pool candidate form status field."
  },
  "Bu20w5": {
    "defaultMessage": "Visible Minority",
    "description": "Title displayed for the Pool Candidates table Visible Minority column."
  },
  "CQLx6F": {
    "defaultMessage": "Visit Canada.ca.",
    "description": "Title for the Canada logo in the Footer."
  },
  "CcOlo0": {
    "defaultMessage": "Key",
    "description": "Title displayed for the CMO Asset table Key column."
  },
  "CdXStL": {
    "defaultMessage": "Telephone",
    "description": "Title displayed on the Pool Candidates table telephone column."
  },
  "Cr+vaa": {
    "defaultMessage": "Name (French):",
    "description": "Label displayed on the pool form name (French) field."
  },
  "D2DUgE": {
    "defaultMessage": "Select one or more CMO Assets...",
    "description": "Placeholder displayed on the pool candidate form cmo assets field."
  },
  "D753gS": {
    "defaultMessage": "Edit",
    "description": "Title displayed for the Classification table Edit column."
  },
  "D8Pgbs": {
    "defaultMessage": "Create Classification",
    "description": "Title displayed on the create a classification form."
  },
  "DexZJJ": {
    "defaultMessage": "Create Classification",
    "description": "Heading displayed above the Create Classification form."
  },
  "DguVoT": {
    "defaultMessage": "Update User",
    "description": "Title displayed on the update a user form."
  },
  "Dx0c3b": {
    "defaultMessage": "Date Modified",
    "description": "Header for the sites last date modification found in the footer."
  },
  "ECIFDU": {
    "defaultMessage": "Pool Candidate updated successfully!",
    "description": "Message displayed to user after pool candidate is updated successfully."
  },
  "EOYrwu": {
    "defaultMessage": "Location Preferences:",
    "description": "Label displayed on the pool candidate form location preferences field."
  },
  "EUqcab": {
    "defaultMessage": "Admin Tools",
    "description": "Label displayed on the Admin Tools menu item."
  },
  "EW/5jT": {
    "defaultMessage": "Owner:",
    "description": "Label displayed on the pool form owner field."
  },
  "EfRVfk": {
    "defaultMessage": "Error: creating user failed",
    "description": "Message displayed to user after user fails to get created."
  },
  "Epj3xB": {
    "defaultMessage": "Email:",
    "description": "Label displayed on the user form email field."
  },
  "ExHECs": {
    "defaultMessage": "Asset Criteria:",
    "description": "Label displayed on the pool form asset criteria field."
  },
  "FGUGtr": {
    "defaultMessage": "Group and Level",
    "description": "Title displayed for the Pool table Group and Level column."
  },
  "FvnSwn": {
    "defaultMessage": "ID",
    "description": "Title displayed on the CMO Asset table ID column."
  },
  "GQ8kKU": {
    "defaultMessage": "Update Operational Requirement",
    "description": "Title displayed on the update a operational requirement form."
  },
  "GTR9Pt": {
    "defaultMessage": "Department updated successfully!",
    "description": "Message displayed to user after department is updated successfully."
  },
  "H5EJq1": {
    "defaultMessage": "Update Pool",
    "description": "Title displayed on the update a pool form."
  },
  "HAEmDX": {
    "defaultMessage": "CMO Assets",
    "description": "Heading displayed above the CMO Asset Table component."
  },
  "HH7lQo": {
    "defaultMessage": "Error: updating pool candidate failed",
    "description": "Message displayed to pool candidate after pool candidate fails to get updated."
  },
  "HUCIzc": {
    "defaultMessage": "Name",
    "description": "Title displayed for the Classification table Name column."
  },
  "HocLRh": {
    "defaultMessage": "Pool Name",
    "description": "Title displayed for the Pool table pool name column."
  },
  "HqXcmo": {
    "defaultMessage": "Operational Requirement updated successfully!",
    "description": "Message displayed to user after operational requirement is updated successfully."
  },
  "IIX75o": {
    "defaultMessage": "Diploma",
    "description": "Title displayed for the Pool Candidates table Diploma column."
  },
  "If8CeH": {
    "defaultMessage": "ID",
    "description": "Title displayed on the Pool Candidates table ID column."
  },
  "IvK9RL": {
    "defaultMessage": "Select one or more classifications...",
    "description": "Placeholder displayed on the pool candidate form expected classifications field."
  },
  "JNYig6": {
    "defaultMessage": "Preferred Language:",
    "description": "Label displayed on the user form preferred language field."
  },
  "Jle35M": {
    "defaultMessage": "Has Diploma:",
    "description": "Label displayed on the pool candidate form has diploma field."
  },
  "KSNNgE": {
    "defaultMessage": "Update Department",
    "description": "Title displayed on the update a department form."
  },
  "Kdfiu3": {
    "defaultMessage": "CMO Asset updated successfully!",
    "description": "Message displayed to user after cmo asset is updated successfully."
  },
  "Ke0TPJ": {
    "defaultMessage": "Maximum Salary",
    "description": "Title displayed for the Classification table Maximum Salary column."
  },
  "LAPK86": {
    "defaultMessage": "Department #:",
    "description": "Label displayed on the create a department form department number field."
  },
  "LCauA2": {
    "defaultMessage": "Name (French):",
    "description": "Label displayed on the operational requirement form name (French) field."
  },
  "LEVK8x": {
    "defaultMessage": "Error: updating classification failed",
    "description": "Message displayed to user after classification fails to get updated."
  },
  "MMP0XC": {
    "defaultMessage": "CMO Assets",
    "description": "Label displayed on the CMO Assets menu item."
  },
  "MpB5zS": {
    "defaultMessage": "Create CMO Asset",
    "description": "Title displayed on the create a cmo asset form."
  },
  "MxuMT5": {
    "defaultMessage": "View our privacy policy.",
    "description": "Title for the privacy link in the Footer."
  },
  "N67tO0": {
    "defaultMessage": "Last Name",
    "description": "Title displayed for the User table Last Name column."
  },
  "NWkoqC": {
    "defaultMessage": "Expiry Date:",
    "description": "Label displayed on the pool candidate form expiry date field."
  },
  "NZyO8h": {
    "defaultMessage": "Indigenous",
    "description": "Title displayed for the Pool Candidates table Indigenous column."
  },
  "NbEyIu": {
    "defaultMessage": "Description",
    "description": "Title displayed for the Operational Requirement table Description column."
  },
  "NhQGh5": {
    "defaultMessage": "Error: creating cmo asset failed",
    "description": "Message displayed to user after cmo asset fails to get created."
  },
  "Nhto6b": {
    "defaultMessage": "CMO Asset created successfully!",
    "description": "Message displayed to user after cmo asset is created successfully."
  },
  "OpKsT/": {
    "defaultMessage": "Language Ability:",
    "description": "Label displayed on the pool candidate form language ability field."
  },
  "OqGhl+": {
    "defaultMessage": "Select a level...",
    "description": "Placeholder displayed on the classification form level field."
  },
  "P5iow/": {
    "defaultMessage": "Key",
    "description": "Title displayed for the Operational Requirement table Key column."
  },
  "QCTZUv": {
    "defaultMessage": "Search",
    "description": "Label displayed on the Global Filter form Search field."
  },
  "QOvS1b": {
    "defaultMessage": "Department #",
    "description": "Title displayed for the Department table Department # column."
  },
  "QPZ1uK": {
    "defaultMessage": "Name FR:",
    "description": "Label displayed on the create a cmo asset form name (French) field."
  },
  "RLdAmI": {
    "defaultMessage": "Group:",
    "description": "Label displayed for the classification form group field."
  },
  "SD2etQ": {
    "defaultMessage": "Classifications",
    "description": "Label displayed on the Classifications menu item."
  },
  "SL1nEE": {
    "defaultMessage": "Essential Criteria:",
    "description": "Label displayed on the pool form essential criteria field."
  },
  "SaAylk": {
    "defaultMessage": "Operational Requirements",
    "description": "Label displayed on the Operational Requirements menu item."
  },
  "Sb2fEr": {
    "defaultMessage": "Pool {poolId} not found.",
    "description": "Message displayed for pool not found."
  },
  "SeyF9s": {
    "defaultMessage": "Pool Description",
    "description": "Title displayed for the Pool table pool description column."
  },
  "SqZuQS": {
    "defaultMessage": "Create Pool Candidate",
    "description": "Title displayed on the create a user form."
  },
  "TAyq5Q": {
    "defaultMessage": "Disability",
    "description": "Title displayed for the Pool Candidates table Disability column."
  },
  "U+WqrO": {
    "defaultMessage": "Update Classification",
    "description": "Title displayed on the update a classification form."
  },
  "UdrVsT": {
    "defaultMessage": "CMO Identifier:",
    "description": "Label displayed on the pool candidate form cmo identifier field."
  },
  "VK2qKI": {
    "defaultMessage": "Users",
    "description": "Heading displayed above the User Table component."
  },
  "VLtsEh": {
    "defaultMessage": "Classifications:",
    "description": "Label displayed on the pool form classifications field."
  },
  "VaVo2t": {
    "defaultMessage": "Error: creating department failed",
    "description": "Message displayed to user after department fails to get created."
  },
  "VcOlXA": {
    "defaultMessage": "Privacy Policy",
    "description": "Label for the privacy link in the Footer."
  },
  "VgbJiw": {
    "defaultMessage": "Owner",
    "description": "Title displayed for the Pool table owner email column."
  },
  "VqyL+/": {
    "defaultMessage": "ID",
    "description": "Title displayed on the Classification table ID column."
  },
  "W2qRX5": {
    "defaultMessage": "Error: creating pool failed",
    "description": "Message displayed to pool after pool fails to get created."
  },
  "WuKWNr": {
    "defaultMessage": "Create Operational Requirement",
    "description": "Title displayed on the create a operational requirement form."
  },
  "XBY4Fq": {
    "defaultMessage": "Create Department",
    "description": "Title displayed on the create a department form."
  },
  "XMGjMX": {
    "defaultMessage": "Minimum Salary:",
    "description": "Label displayed for the classification form min salary field."
  },
  "YO9JAn": {
    "defaultMessage": "Select a language ability...",
    "description": "Placeholder displayed on the pool candidate form language ability field."
  },
  "YOHJFE": {
    "defaultMessage": "Error: creating pool candidate failed",
    "description": "Message displayed to pool candidate after pool candidate fails to get created."
  },
  "YWhS9r": {
    "defaultMessage": "CMO Assets:",
    "description": "Label displayed on the pool candidate form cmo assets field."
  },
  "ZE77nf": {
    "defaultMessage": "Canada.ca",
    "description": "Label for the Canada link in the Footer."
  },
  "ZGpncy": {
    "defaultMessage": "Terms & Conditions",
    "description": "Label for the terms and conditions link in the Footer."
  },
  "ZZaEho": {
    "defaultMessage": "Description",
    "description": "Title displayed for the CMO Asset table Description column."
  },
  "ZbW9T7": {
    "defaultMessage": "Error: updating pool failed",
    "description": "Message displayed to pool after pool fails to get updated."
  },
  "ZbpbD6": {
    "defaultMessage": "Create Department",
    "description": "Heading displayed above the Create Department form."
  },
  "ZumF63": {
    "defaultMessage": "Canada's Logo.",
    "description": "Alt text for the Canada logo in the Footer."
  },
  "a48ztb": {
    "defaultMessage": "Error: creating operational requirement failed",
    "description": "Message displayed to user after operational requirement fails to get created."
  },
  "aGEisH": {
    "defaultMessage": "Edit",
    "description": "Title displayed for the Pool Candidates table Edit column."
  },
  "aS4Lty": {
    "defaultMessage": "Group",
    "description": "Title displayed for the Classification table Group column."
  },
  "adfUDi": {
    "defaultMessage": "Operational Requirement created successfully!",
    "description": "Message displayed to user after operational requirement is created successfully."
  },
  "b3VnhM": {
    "defaultMessage": "Classification {classificationId} not found.",
    "description": "Message displayed for classification not found."
  },
  "bhl2kK": {
    "defaultMessage": "Key:",
    "description": "Label displayed on the classification form key field."
  },
  "bxh7Ba": {
    "defaultMessage": "Select one or more location preferences...",
    "description": "Placeholder displayed on the pool candidate form location preferences field."
  },
  "cT/8aw": {
    "defaultMessage": "Select one or more expected salaries...",
    "description": "Placeholder displayed on the pool candidate form expected salary field."
  },
  "cy2UdP": {
    "defaultMessage": "Update Pool Candidate",
    "description": "Title displayed on the update a user form."
  },
  "dJ4aWF": {
    "defaultMessage": "Edit",
    "description": "Title displayed for the Operational Requirement table Edit column."
  },
  "ekKb+G": {
    "defaultMessage": "Create Pool",
    "description": "Title displayed on the create a pool form."
  },
  "evxvnW": {
    "defaultMessage": "User updated successfully!",
    "description": "Message displayed to user after user is updated successfully."
  },
  "f9E/6Q": {
    "defaultMessage": "Select one or more operational requirements...",
    "description": "Placeholder displayed on the pool candidate form operational requirements field."
  },
  "fAIozk": {
    "defaultMessage": "User created successfully!",
    "description": "Message displayed to user after user is created successfully."
  },
  "fXMsoK": {
    "defaultMessage": "Telephone",
    "description": "Title displayed for the User table Telephone column."
  },
  "g0PYPY": {
    "defaultMessage": "Maximum Salary:",
    "description": "Label displayed for the classification form max salary field."
  },
  "h6kyvS": {
    "defaultMessage": "Select a user...",
    "description": "Placeholder displayed on the pool candidate form user field."
  },
  "hTfHUv": {
    "defaultMessage": "Edit",
    "description": "Title displayed for the Department table Edit column."
  },
  "hnyrdl": {
    "defaultMessage": "Classifications",
    "description": "Heading displayed above the Classification Table component."
  },
  "iAiwKZ": {
    "defaultMessage": "Description (English):",
    "description": "Label displayed on the operational requirement form description (English) field."
  },
  "inCoGo": {
    "defaultMessage": "Pool Candidates",
    "description": "Label displayed on the Pool Candidates menu item."
  },
  "iu8ezY": {
    "defaultMessage": "First Name:",
    "description": "Label displayed on the user form first name field."
  },
  "jJCDGc": {
    "defaultMessage": "Classification updated successfully!",
    "description": "Message displayed to user after classification is updated successfully."
  },
  "jKxpPh": {
    "defaultMessage": "Pool:",
    "description": "Label displayed on the pool candidate form pool field."
  },
  "jRZVBS": {
    "defaultMessage": "Name (French):",
    "description": "Label displayed on the classification form name (French) field."
  },
  "jWVB8I": {
    "defaultMessage": "Preferred Lang",
    "description": "Title displayed on the Pool Candidates table Preferred Lang column."
  },
  "kt5dPm": {
    "defaultMessage": "Id",
    "description": "Title displayed on the Pool table Unique Identifier column."
  },
  "l7xk8j": {
    "defaultMessage": "Select one or more asset...",
    "description": "Placeholder displayed on the pool form asset criteria field."
  },
  "lZ/P9m": {
    "defaultMessage": "Woman:",
    "description": "Label displayed on the pool candidate form is woman field."
  },
  "mZMdKM": {
    "defaultMessage": "Submit feedback to GC Talent Cloud via email.",
    "description": "Title for the feedback link in the Footer."
  },
  "mbxksT": {
    "defaultMessage": "First Name",
    "description": "Title displayed on the User table First Name column."
  },
  "mf+QEY": {
    "defaultMessage": "Preferred Language",
    "description": "Title displayed for the User table Preferred Language column."
  },
  "mjbSrx": {
    "defaultMessage": "Error: updating operational requirement failed",
    "description": "Message displayed to user after operational requirement fails to get updated."
  },
  "n0pfBx": {
    "defaultMessage": "Pool",
    "description": "Title displayed for the Pool Candidates table Pool column."
  },
  "nXRLAX": {
    "defaultMessage": "Error: updating department failed",
    "description": "Message displayed to user after department fails to get updated."
  },
  "oOnQMV": {
    "defaultMessage": "Name (English):",
    "description": "Label displayed on the classification form name (English) field."
  },
  "os4TSk": {
    "defaultMessage": "Select an owner...",
    "description": "Placeholder displayed on the pool form owner field."
  },
  "pBuPWx": {
    "defaultMessage": "Canada.ca",
    "description": "Label for the Canada logo in the Footer."
  },
  "qL43Gw": {
    "defaultMessage": "Pools",
    "description": "Heading displayed above the Pool Table component."
  },
  "qYH0du": {
    "defaultMessage": "Edit",
    "description": "Title displayed for the User table Edit column."
  },
  "rC5nGg": {
    "defaultMessage": "Name EN:",
    "description": "Label displayed on the create a cmo asset form name (English) field."
  },
  "rE/ytf": {
    "defaultMessage": "CMO Asset {cmoAssetId} not found.",
    "description": "Message displayed for cmo asset not found."
  },
  "rY6HZD": {
    "defaultMessage": "Has Disability:",
    "description": "Label displayed on the pool candidate form has disability field."
  },
  "tKs7dH": {
    "defaultMessage": "Description (English):",
    "description": "Label displayed on the pool form description (English) field."
  },
  "tNLMAs": {
    "defaultMessage": "Expected Salary:",
    "description": "Label displayed on the pool candidate form expected salary field."
  },
  "tcY5cG": {
    "defaultMessage": "Name",
    "description": "Title displayed for the Operational Requirement table Name column."
  },
  "tfULo6": {
    "defaultMessage": "Operational Requirements",
    "description": "Heading displayed above the Operational Requirement Table component."
  },
  "tpzt/B": {
    "defaultMessage": "Edit",
    "description": "Title displayed for the Pool table Edit column."
  },
  "uZYlhm": {
    "defaultMessage": "Operational Requirements:",
    "description": "Label displayed on the pool candidate form operational requirements field."
  },
  "uc/NCf": {
    "defaultMessage": "Language",
    "description": "Title displayed for the Pool Candidates table Language column."
  },
  "vJCxlc": {
    "defaultMessage": "Expected Classifications:",
    "description": "Label displayed on the pool candidate form expected classifications field."
  },
  "vkus5u": {
    "defaultMessage": "Select one or more operational requirements...",
    "description": "Placeholder displayed on the pool form operational requirements field."
  },
  "vnhTgE": {
    "defaultMessage": "Select a language...",
    "description": "Placeholder displayed on the user form preferred language field."
  },
  "w42ZU3": {
    "defaultMessage": "Operational Requirement {operationalRequirementId} not found.",
    "description": "Message displayed for operational requirement not found."
  },
  "wCBE9S": {
    "defaultMessage": "Pools",
    "description": "Label displayed on the Pools menu item."
  },
  "wZ91g+": {
    "defaultMessage": "Pool created successfully!",
    "description": "Message displayed to user after pool is created successfully."
  },
  "wdzz8g": {
    "defaultMessage": "Last Name:",
    "description": "Label displayed on the user form last name field."
  },
  "x+2gYv": {
    "defaultMessage": "Create Operational Requirement",
    "description": "Heading displayed above the Create Operational Requirement form."
  },
  "xAjes8": {
    "defaultMessage": "Hide/Show Table Columns",
    "description": "Label displayed on the Table Columns toggle."
  },
  "xxLs3C": {
    "defaultMessage": "Name",
    "description": "Title displayed on the Pool Candidates table name column."
  },
  "yGlG9e": {
    "defaultMessage": "Department created successfully!",
    "description": "Message displayed to user after department is created successfully."
  },
  "yZqUAU": {
    "defaultMessage": "Level",
    "description": "Title displayed for the Classification table Level column."
  },
  "yoshF7": {
    "defaultMessage": "Pool Candidate created successfully!",
    "description": "Message displayed to user after pool candidate is created successfully."
  },
  "z+TEpN": {
    "defaultMessage": "Create Pool Candidate",
    "description": "Heading displayed above the Create Pool Candidate form."
  },
  "z2m2Gp": {
    "defaultMessage": "Edit",
    "description": "Title displayed for the CMO Asset table Edit column."
  }
}
