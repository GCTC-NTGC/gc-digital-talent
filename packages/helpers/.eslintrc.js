module.exports = {
  root: true,
  extends: ["custom"],
  plugins: ["formatjs"],
  rules: {
    // this package does not have a translation script
    "formatjs/blocklist-elements": ["error", ["literal"]],
    "import/no-unused-modules": [1, { unusedExports: true, ignoreExports: ["src/index.{ts,tsx}"] }],
  },
};
