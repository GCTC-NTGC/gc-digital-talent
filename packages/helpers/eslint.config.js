import baseConfig from "@gc-digital-talent/eslint-config/no-translation";

export default [
  ...baseConfig,
  {
    rules: {
      "import/no-unused-modules": (() => {
        // Trigger warning on CI only, because of performance issues locally
        if (process.env.CI) {
          return [
            1,
            {
              unusedExports: true,
              ignoreExports: ["src/index.{ts,tsx}"],
            },
          ];
        }
        return "off";
      })(),
    },
  },
];
