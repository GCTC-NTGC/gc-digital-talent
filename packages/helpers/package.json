{
  "name": "@gc-digital-talent/helpers",
  "license": "AGPL-3.0",
  "version": "0.0.0",
  "types": "./src/index.tsx",
  "exports": {
    ".": "./src/index.tsx"
  },
  "scripts": {
    "lint": "eslint \"src/**/*.ts*\"",
    "test": "jest --detectOpenHandles --forceExit --passWithNoTests"
  },
  "dependencies": {
    "@gc-digital-talent/graphql": "workspace:*",
    "@gc-digital-talent/storage": "workspace:*",
    "graphql": "^16.8.1",
    "lodash": "^4.17.21",
    "react": "^18.2.0",
    "urql": "^4.0.7"
  },
  "devDependencies": {
    "@gc-digital-talent/eslint-config": "workspace:*",
    "@types/lodash": "^4.17.0",
    "@types/react": "^18.2.78",
    "eslint": "^9.1.1",
    "tsconfig": "workspace:*"
  }
}
