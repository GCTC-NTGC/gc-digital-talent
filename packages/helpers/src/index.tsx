import {
  keyStringRegex,
  phoneNumberRegex,
  workEmailDomainRegex,
} from "./constants/regularExpressions";
import buildMailToUri from "./utils/buildMailToUri";
import normalizeString from "./utils/normalizeString";
import sanitizeUrl from "./utils/sanitizeUrl";
import isUuidError from "./utils/uuid";
import {
  assertUnreachable,
  notEmpty,
  empty,
  getId,
  hasKey,
  getOrThrowError,
  deleteProperty,
  insertBetween,
  emptyToNull,
  emptyToUndefined,
  uniqueItems,
  groupBy,
  pickMap,
  unpackMaybes,
  localizedEnumHasValue,
} from "./utils/util";
import useIsSmallScreen from "./hooks/useIsSmallScreen";
import { GraphqlType } from "./types/graphql";
import { NotFoundError, UnauthorizedError } from "./errors";

export {
  assertUnreachable,
  keyStringRegex,
  phoneNumberRegex,
  workEmailDomainRegex,
  notEmpty,
  empty,
  getId,
  hasKey,
  getOrThrowError,
  deleteProperty,
  insertBetween,
  emptyToNull,
  emptyToUndefined,
  uniqueItems,
  buildMailToUri,
  normalizeString,
  sanitizeUrl,
  isUuidError,
  useIsSmallScreen,
  groupBy,
  pickMap,
  unpackMaybes,
  localizedEnumHasValue,
  NotFoundError,
  UnauthorizedError,
};
export type { GraphqlType };
