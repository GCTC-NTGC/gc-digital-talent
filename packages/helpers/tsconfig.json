{
  "extends": "tsconfig/react-lib.json",
  "include": [
    "src"
  ],
  "exclude": [
    "dist",
    "build",
    "node_modules"
  ]
}
