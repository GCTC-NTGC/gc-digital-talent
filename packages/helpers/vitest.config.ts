import { defineConfig } from "vitest/config";

import { baseConfig } from "@gc-digital-talent/vitest-helpers/config";

export default defineConfig({
  ...baseConfig,
});
