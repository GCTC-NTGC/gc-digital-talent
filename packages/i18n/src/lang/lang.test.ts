import groupBy from "lodash/groupBy"; // provides sorted output
import stringify from "json-stable-stringify";

import * as rawWebEnMessages from "@gc-digital-talent/web/src/lang/en.json";
import * as rawWebFrMessages from "@gc-digital-talent/web/src/lang/fr.json";
import { notEmpty } from "@gc-digital-talent/helpers";

import * as rawI18nEnMessages from "./en.json";
import * as rawI18nFrMessages from "./fr.json";

describe("message files", () => {
  /*
    To update expected changes to the snapshot:
    `npx jest ./src/lang/lang.test.ts --testNamePattern="should have no changes to duplicate strings" --updateSnapshot`
    */
  it("should have no changes to duplicate strings", async () => {
    const enMessages: Record<string, { defaultMessage: string }> = {
      ...rawWebEnMessages,
      ...rawI18nEnMessages,
    };

    const frMessages: Record<string, { defaultMessage: string }> = {
      ...rawWebFrMessages,
      ...rawI18nFrMessages,
    };

    const matchedMessages = Object.keys(enMessages).map((messageId) => ({
      en: enMessages[messageId].defaultMessage,
      fr: frMessages[messageId]?.defaultMessage,
    }));
    const groupedByEn = groupBy(matchedMessages, "en");
    const messagesWithMultipleFrStrings = Object.keys(groupedByEn)
      .map((en) => ({
        en,
        fr: Array.from(
          new Set(groupedByEn[en].map((g) => g.fr).filter(notEmpty)),
        ),
      }))
      .filter((g) => g.fr.length > 1);

    expect(
      stringify(messagesWithMultipleFrStrings, { space: "  " }),
    ).toMatchSnapshot();
  });
});
