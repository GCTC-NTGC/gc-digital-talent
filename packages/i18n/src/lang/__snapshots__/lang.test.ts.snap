// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message files should have no changes to duplicate strings 1`] = `
"[
    {
        "en": "Education requirement",
        "fr": [
            "Exigence d’études",
            "Exigences pédagogiques"
        ]
    },
    {
        "en": "Career timeline",
        "fr": [
            "Parcours professionnel",
            "Échéancier de carrière"
        ]
    },
    {
        "en": "Screened in",
        "fr": [
            "Sélectionné(e)",
            "Accepté"
        ]
    },
    {
        "en": "Language requirement",
        "fr": [
            "Exigence linguistique",
            "Exigences relatives à la langue"
        ]
    },
    {
        "en": "N/A",
        "fr": [
            "S.O.",
            "S. O."
        ]
    },
    {
        "en": "Minimum education",
        "fr": [
            "Niveau d'études minimal",
            "Études minimales"
        ]
    },
    {
        "en": "View",
        "fr": [
            "Voir",
            "Affichage"
        ]
    },
    {
        "en": "Required",
        "fr": [
            "Requises",
            "Champ obligatoire"
        ]
    },
    {
        "en": "Skill requirements",
        "fr": [
            "Exigences relatives aux compétences",
            "Exigences en matière de compétences"
        ]
    },
    {
        "en": "Optional",
        "fr": [
            "Facultatives",
            "Facultatif"
        ]
    },
    {
        "en": "About this role",
        "fr": [
            "À propos de ce rôle",
            "Au sujet de ce rôle"
        ]
    }
]"
`;
